### PR TITLE
CI: update pre-commit

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Update recommit config and enable CI automation

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,36 +15,45 @@
...
+    rev: 3.9.1
+    hooks:
       - id: flake8
         name: PEP8
         entry: flake8
-        language: system
+        language: python
         types: [python]
-
-  - repo: https://github.com/pre-commit/mirrors-mypy
```
Comment (tchaton) [inline .pre-commit-config.yaml:47]: Why removing mypy ?

Comment (Borda) [reply .pre-commit-config.yaml:47]: good catch :+1:

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,36 +15,46 @@
...
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
+    rev: v3.4.0
     hooks:
-      - id: trailing-whitespace
       - id: end-of-file-fixer
+      - id: trailing-whitespace
+      #- id: check-yaml
+      - id: pretty-format-json
```
Comment (carmocca) [inline .pre-commit-config.yaml:30]: Does the repo have json files?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,36 +15,46 @@
...
+
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
+    rev: v3.4.0
     hooks:
-      - id: trailing-whitespace
       - id: end-of-file-fixer
+      - id: trailing-whitespace
+      #- id: check-yaml
```
Comment (carmocca) [inline .pre-commit-config.yaml:29]: Remove this if not activating it

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -15,36 +15,46 @@
 default_language_version:
   python: python3.8
 
+ci:
+  autofix_prs: true
+  autoupdate_commit_msg: '[pre-commit.ci] pre-commit suggestions'
+  # submodules: true
```
Comment (carmocca) [inline .pre-commit-config.yaml:21]: We have no submodules... remove this?

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -15,36 +15,46 @@
...
+    rev: 5.8.0
     hooks:
       - id: isort
         name: imports
         entry: isort
-        args: [--settings-path, ./pyproject.toml]
-        language: system
+        language: python
         types: [python]
+        require_serial: false
```
Comment (carmocca) [inline .pre-commit-config.yaml:40]: what is `require_serial` for?

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -15,36 +15,46 @@
...
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.8.0
     hooks:
       - id: isort
         name: imports
         entry: isort
-        args: [--settings-path, ./pyproject.toml]
-        language: system
+        language: python
         types: [python]
```
Comment (carmocca) [inline .pre-commit-config.yaml:39]: I don't think we need to annotate:

```
        language: python
        types: [python]
```

But not 100% sure.

Same for the other hooks

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Borda) [reply .pre-commit-config.yaml:30]: GH actions

File: .pre-commit-config.yaml (hunk 5 above)
Comment (Borda) [reply .pre-commit-config.yaml:40]: pre-commit was complaining that it was missing

File: .pre-commit-config.yaml (hunk 6 above)
Comment (Borda) [reply .pre-commit-config.yaml:39]: some required it, probably not all...

File: .pre-commit-config.yaml (hunk 2 above)
Comment (carmocca) [reply .pre-commit-config.yaml:28]: But those are yaml files, not json

Comment (Borda) [reply .pre-commit-config.yaml]: true my bad :D

File: .pre-commit-config.yaml (hunk 5 above)
Comment (carmocca) [reply .pre-commit-config.yaml:36]: According to this:

https://pre-commit.com/#creating-new-hooks

It's already false by default, so shouldn't be necessary

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**